src/hpc/tls.h

```diff
@@ -55,19 +55,40 @@ typedef struct ThreadLocalStorage
   UInt LocksContended;
 } ThreadLocalStorage;
 
-#ifdef HAVE_NATIVE_TLS
+#ifdef USE_NATIVE_TLS
 extern __thread ThreadLocalStorage *TLSInstance;
 #endif
 
+#ifdef USE_PTHREAD_TLS
+
+#include <pthread.h>
+
+#ifdef __GNUC__
+// This trick allows gcc/clang to hoist calls to pthread_getspecific()
```

Comment:
Which "trick" does this refer to? Let's be explicit! Is it the use of `__attribute__((pure)) `? Then let's say that: "Using __attribute__((pure)) here is a trick that allows ..."

Reply:
Reworded.